### ISSUE TITLE
[Tree] Improve focus behavior when tab'ing button in a tree node

### ISSUE THREAD

**Issue by jpzwarte** [sl-design-system/components#2826]
### Provide a general summary of the issue here

When you press the TAB key when you're on the last interactive element in a tree-node (so the buttons in this exmaple: https://storybook.sanomalearning.design/?path=/story/navigation-tree--buttons), focus jumps outside of the tree. This is unwanted behavior. However, it's not clear how it should behave. We need to investigate here how other tree components handle this.

### 👤 Your name

Jeroen

### 🧢 Your Product/Team

SLDS
